src/engine/Union.cpp

```diff
@@ -283,3 +283,11 @@ Result::Generator Union::computeResultLazily(
     }
   }
 }
+
+// _____________________________________________________________________________
+std::unique_ptr<Operation> Union::clone() const {
+  auto copy = std::make_unique<Union>(*this);
+  copy->_subtrees.at(0) = _subtrees.at(0)->clone();
+  copy->_subtrees.at(1) = _subtrees.at(1)->clone();
```

Comment:
Make this a for-loop, s.t. it doesn't automatically and silently break once we add e.g. a k-way union.